src/rp2_common/pico_rand/include/pico/rand.h

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) 2022 Raspberry Pi (Trading) Ltd.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#ifndef _PICO_RAND_H_
+#define _PICO_RAND_H_
+
+#include "pico.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** \file pico/rand.h
+ *  \defgroup pico_rand pico_rand
+ *
+ * Random Number Generator API
+ *
+ */
+
+// The (Pseudo) Random Number Generator component holds 128 bits of state
+struct prng_128_t {
+    uint64_t s[2];
+};
+
+#define PICO_RAND_SEED_SIZE_BYTES       sizeof(struct prng_128_t)
+
+
+/*! \brief Initialise Pico Random Number Generator
+ *  \ingroup pico_rand
+ *
+ * This function should be called during application initialisation.
+ * It will take about 80ms to execute because the entire RAM is used
+ * to generate a hash as part of the seed generation the first time it
+ * is called.
+ *
+ * The seed argument is optional since the rand module will generate a
+ * seed in all cases.  If a user seed _is_ supplied, it will be XORed
+ * with the locally generated seed to produce the initial value used
+ * to start the PRNG.
+ *
+ * This function can be called as often as required.  Note: If the RAM has
+ * already been hashed, either becuase this function has previously been called
+ * or a random number has been already requested, it is optional to hash the
+ * RAM contents again, not doing so reduces the re-seeding time to ~1ms.
+ *
+ * NOTE:  Code within the rand module will automatically call this function
+ * the first time a RN is requested *if* it has not previously been called.
```

Comment:
Presumably 'RN' here stands for Random Number? Might be better to avoid the abbreviation, for the sake of clarity.